### DISCUSSION TITLE
Midi clip creation

### DISCUSSION THREAD

**Discussion by gilesbutler** [djipco/webmidi#303] [Q&A] [answered]
Hi there,

Awesome library and thanks so much for your work.

I'm trying to create a structure like the one below but not sure where to start or if it's even possible?

I'd like to create either empty midi clips or it's ok if they have one long note in for the length of the clip.

And I'd like the ability to add space before and after the midi clips.

I also need to be able to save the individual tracks as midi files as well all the tracks collectively as one midi file.

Is any/all of this possible? Can anybody point me in the right direction on where to start please? 😄

Thanks

<img width="459" alt="200269747-0f3cb1dd-1b97-4811-a6dd-233d1992f633" src="https://user-images.githubusercontent.com/790734/200771716-59c723d4-4ba9-480d-b95b-b8aafdb585e8.png">

**Comment by djipco** [comment, marked as answer]
This is outside the purview of WEBMIDI.js. However, there are [various libraries](https://www.npmjs.com/search?q=midi%20file) that can read and write MIDI files. Perhaps this is one place to start from?

**Reply by gilesbutler** [reply]
Thanks for your reply @djipco, I'll check out the other libraries
